tests/cpp_integration_tests/test_main_model.cpp

```diff
@@ -1373,67 +1374,117 @@ TEST_CASE("Test main model - runtime dispatch") {
-            MainModel columnar_model{base_model};
-            row_based_model.update_component<permanent_update_t>(update_data_with_rows);
-            columnar_model.update_component<permanent_update_t>(update_data_with_columns);
-
-            std::vector<SymNodeOutput> node_output_from_base(state.node_input.size());
-            std::vector<SymNodeOutput> node_output_from_row_based(state.node_input.size());
-            std::vector<SymNodeOutput> node_output_from_columnar(state.node_input.size());
-
-            MutableDataset sym_output_from_base{true, 1, "sym_output", meta_data::meta_data_gen::meta_data};
-            sym_output_from_base.add_buffer("node", node_output_from_base.size(), node_output_from_base.size(), nullptr,
-                                            node_output_from_base.data());
-            MutableDataset sym_output_from_row_based{true, 1, "sym_output", meta_data::meta_data_gen::meta_data};
-            sym_output_from_row_based.add_buffer("node", node_output_from_row_based.size(),
-                                                 node_output_from_row_based.size(), nullptr,
-                                                 node_output_from_row_based.data());
-            MutableDataset sym_output_from_columnar{true, 1, "sym_output", meta_data::meta_data_gen::meta_data};
-            sym_output_from_columnar.add_buffer("node", node_output_from_columnar.size(),
-                                                node_output_from_columnar.size(), nullptr,
-                                                node_output_from_columnar.data());
-
-            base_model.calculate(options, sym_output_from_base);
-            row_based_model.calculate(options, sym_output_from_row_based);
-            columnar_model.calculate(options, sym_output_from_columnar);
-
-            REQUIRE(node_output_from_columnar.size() == node_output_from_base.size());
-            REQUIRE(node_output_from_columnar.size() == node_output_from_row_based.size());
+            std::ranges::transform(state.sym_load_update, std::back_inserter(sym_load_ids),
+                                   [](auto const& sym_load) { return sym_load.id; });
+            REQUIRE(sym_load_ids.size() == sym_load_p_specified.size());
+            REQUIRE(sym_load_p_specified.size() == state.sym_load_update.size());
+
+            SUBCASE("With IDs") {
+                ConstDataset update_data_with_rows{false, 1, "update", meta_data::meta_data_gen::meta_data};
+                update_data_with_rows.add_buffer("sym_load", state.sym_load_update.size(), state.sym_load_update.size(),
+                                                 nullptr, state.sym_load_update.data());
+
+                ConstDataset update_data_with_columns{false, 1, "update", meta_data::meta_data_gen::meta_data};
+                update_data_with_columns.add_buffer("sym_load", sym_load_p_specified.size(),
+                                                    sym_load_p_specified.size(), nullptr, nullptr);
```

Comment:
for the sake of readability (and since you don't re-use this for the case without IDs), please revert this line
```suggestion
                update_data_with_columns.add_buffer("sym_load", sym_load_ids.size(),
                                                    sym_load_ids.size(), nullptr, nullptr);
```

Reply:
I am not sure I follow the argument.

Reply:
for anyone reading this test, it's more natural to think in terms of ID-major. hence, the size would be given by the `sym_load_ids`.

Only in the very specific case that the IDs are not provided does it make sense not to think in terms of ID-major

Reply:
Addressed.